### ISSUE TITLE
Distance transform: learn more

### ISSUE THREAD

**Issue by tischi** [NEUBIAS/training-resources#267]
@dlegland Do you know a good article where the Chamfer distances are explained?

**Comment by dlegland** [comment]
Hi @tischi ,

I usually refer to these ones:
* Borgefors, G. (1984). Distance transformation in arbitrary dimensions. Comp. Vis. Graph. Im. Proc., 27, 321–145.
* Borgefors, G. (1986). Distance transformations in digital images. Comp. Vis. Graph. Im. Proc., 34, 344–371.
Somewhat old, but well illustrated!

Some illustrations in MorphoLibJ user manual as well.

**Comment by tischi** [comment]
Thanks @dlegland would it be OK for you if I copy below text and figure into the training material?

<img width="500" alt="image" src="https://user-images.githubusercontent.com/2157566/152333412-f7895c43-6001-425a-95b3-3daf41a89260.png">

It would end up somewhere on this website: https://neubias.github.io/training-resources/distance_transform/index.html

**Comment by dlegland** [comment]
yes, this is fine for me!

**Comment by tischi** [comment]
Thanks!

In fact, I don't get it, @dlegland. In above figure, how would you compute the actual distances? I guess you have to still do something, like divide by something? Is this the "normalize" option? 

It would be great to have beneath another panel with the actual distances that are computed using those weights.

**Comment by dlegland** [comment]
Yes, the "normalize" option consists in dividing all values (well, all values within the foreground) by the weight associated to orthogonal neighbors (the first value in the parens for each sub-panel).

Adding another panel could be nice, I agree, but one should also manage integer vs floating pint outputs (the resulting precision would not be the same...). This may lead to too many figures in the user manual... But maybe as additional material (appendices, or companion slides about chamfer distance transform?).

**Comment by tischi** [comment]
Thanks!

And how does the algorithm work if you have more than one background pixel?
Does it resolve conflicts by taking the minimum distance?

And, is it correct to say that the point of the Chamfer distances is that you can compute them by building up the distances and only have to inspect nearest neighbors O(N)? Whereas for true Euclidian distances you may have to loop for each foreground pixel through the whole image to find the closest background pixel O(N^2)?
